### DISCUSSION TITLE
Please don't deprecate cs.github.com!

### DISCUSSION THREAD

**Discussion by AlexDev404** [community/community#50718] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I would even pay for this experience if it's because of the maintainability costs that is causing it to be shut down as CS it isn't like GH-Search which just gives general results as instead CS is really specific in what it highlights along with giving you the ability to explore the code with a simple UI without ever having to leave the page. Please reconsider keeping this feature alive as it's something I and others really rely on when browsing code.

I tried out GH-Search many times and for me it's more like just a generalized search for when you're not really looking for code but more the entire of GitHub. For me, Code Search and GitHub Search are two entirely different species and I get the idea of having the code search functionality embedded into GitHub and how it unifies things but having a code-search-only type of functionality is something that I really value as the UI and all that doesnt matter as much as having something that you can always drop to when you only want to search for only code and only want tools relating to searching code made available to you without the distractions.

There isn't any other good place on earth to search for only code and IMO CS is the only _actual_ code-only search engine that gives code-only results along with browsing code-only results without having to redirect to some other UI and it really hurts a lot to see it go and I really dont want that which is why I'm even willing to pay a fee to use it if it has to come down to that.

At least give it a chance!!! T_T

I really think CS can be a great thing as many people still use it and love it as I do and I'm sure like me many people would be willing to pay to continue using it and sharing the site with others. I'd do anything to not see this feature leave and I really dont want it to die (please!). Also unlike GitHub's main UI it's easier to share files of code when you only want to view only the code. [Example](https://cs.github.com/StarOfService/jitsi-meet/blob/61a674883472d427cdf8f16a1e08648350a9ccfb/modules/UI/authentication/AuthHandler.js?q=xmpp+language%3AJavaScript#L10) VS [This](https://github.com/StarOfService/jitsi-meet/blob/61a674883472d427cdf8f16a1e08648350a9ccfb/modules/UI/authentication/AuthHandler.js?q=xmpp+language%3AJavaScript#L10). Isn't it easier to view? The GH-CS view seems more like for a purpose when you're saying something like **"Look at this file"** versus **"Look at this directory"** as CS doesn't show things that you don't need to see when you're browsing code. I get the need to place it in the main GitHub UI as you may need those things when managing a repository, but when you only need to search through code there are only a couple of things you may need (basic file browsing, highlighting and a couple search tools).

That's what I meant when I was saying that they are like animals of two different species because GH-Search's functionality is more feature-packed because it's more for like "Search anything and everything" as it was built for more general use compared to CS's functionality it provides that is more focusing and specific as it was built for a specific purpose and more for like "Search through code".

Please reconsider and thanks in advance. I'd really not wanna see this feature die because I seriously don't know what to do without it since I use it regularly when I get stuck coding or in learning a new language.

CS is a whole other world for me in terms of search functionality and in all honesty by just using it to search through code and examples I even learnt Ruby on the way! The precise ability it gives to search through any and every single line of code and the focus ability it provides is incomparable to what a traditional global search functionality would give. It allows you to view every comment and every personal note that developers leave for themselves to remember. And some of those are really useful when it comes to solving a problem or finding and fixing a bug. Some of these things aren't even published or made available on Stack Overflow and the internet.

Please reconsider :(
and thanks I guess

**Comment by colinwm** [comment]
Just wanted to make sure, but are you using the new code search in github.com? I.e. this one: https://github.com/features/code-search

**Reply by AlexDev404** [reply]
Yes, that is the one I am using.
